feat: add `blas/base/icamax`

ShabiShett07 opened the pull request:
Progresses #2039

## Description

> What is the purpose of this pull request?

This pull request:

-   includes the support for Javascript and obtains Fortran reference from [this page](https://www.netlib.org/lapack/explore-html/dd/d52/group__iamax_gafdf273dcc3f020e2aa5c716c1b3d7265.html)

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2039 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

ShabiShett07 committed:
feat: add blas/base/icamax

ShabiShett07 committed:
docs: files completed

ShabiShett07 commented:
/stdlib lint-autofix

ShabiShett07 commented:
/stdlib update-copyright-years

stdlib-bot committed:
fix: resolve lint errors

ShabiShett07 commented:
@kgryte, Please do review this!

aman-095 commented:
/stdlib merge

aman-095 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

aman-095 commented:
Use `Progresses` in the description, `Resolves` will automatically close the issue associated with #2039 once this PR gets merged.

aman-095 committed:
Update package.json

Check for the indentation and brackets. 

Signed-off-by: Aman Bhansali <92033532+aman-095@users.noreply.github.com>

ShabiShett07 commented:
> Use `Progresses` in the description, `Resolves` will automatically close the issue associated with #2039 once this PR gets merged.

Thank you Aman sir for making the necessary changes in this PR.

May I know is this PR ready for merge?

aman-095 committed:
Update README.md

Signed-off-by: Aman Bhansali <92033532+aman-095@users.noreply.github.com>

aman-095 reviewed (changes requested):
Update variable for index from `icx` to `idx`.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/README.md:
```diff
@@ -0,0 +1,172 @@
+
+[blas]: http://www.netlib.org/blas
+
+[icamax]: https://netlib.org/lapack/explore-html/d0/da5/izamax_8f.html
```
Update the link here it's for `izamax`.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/benchmark/benchmark.js:
```diff
@@ -0,0 +1,105 @@
+* @returns {Function} benchmark function
+*/
+function createBenchmark( len ) {
+	var cx;
+
+	cx = new Complex64Array( uniform( len*2, -100.0, 100.0, options ) );
```
`var cx = new Complex64Array( uniform( len*2, -100.0, 100.0, options ) );`

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/benchmark/benchmark.js:
```diff
@@ -0,0 +1,105 @@
+	* @param {Benchmark} b - benchmark instance
+	*/
+	function benchmark( b ) {
+		var icx;
```
Change `icx` to `idx`, here and elsewhere.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/benchmark/benchmark.ndarray.js:
```diff
@@ -0,0 +1,105 @@
+* @returns {Function} benchmark function
+*/
+function createBenchmark( len ) {
+	var cx;
+
+	cx = new Complex64Array( uniform( len*2, -100.0, 100.0, options ) );
```
Same comment as above.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/benchmark/benchmark.ndarray.js:
```diff
@@ -0,0 +1,105 @@
+
+		b.tic();
+		for ( i = 0; i < b.iterations; i++ ) {
+			icx = icamax( cx.length, cx, 1, 0 );
```
same comment.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/repl.txt:
```diff
@@ -0,0 +1,91 @@
+    Indexing is relative to the first index. To introduce an offset, use typed
+    array views.
+
+    If `N < 1`, the function returns `-1`.
```
If `N < 1` or `strideX <= 0`, the function returns `-1`.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/repl.txt:
```diff
@@ -0,0 +1,91 @@
+
+    Returns
+    -------
+    icx: integer
```
same comment

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/repl.txt:
```diff
@@ -0,0 +1,91 @@
+    --------
+    // Standard Usage:
+    > var cx;
+    > cx = new {{alias:@stdlib/array/complex64}}( [ -2.0, 1.0, 3.0, -5.0, 4.0, 0.0 ] );
```
Change this to `var cx = new ..`

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/repl.txt:
```diff
@@ -0,0 +1,91 @@
+    > var icx = {{alias}}( cx.length, cx, 1 )
+    1
+
+    // Using `N` and `strideX` parameters:
```
`strideX` -- `stride`

aman-095 replied:
Any reason behind not updating this and marking the comment as resolved.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/repl.txt:
```diff
@@ -0,0 +1,91 @@
+
+    Returns
+    -------
+    icx: integer
```
same comment.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/repl.txt:
```diff
@@ -0,0 +1,91 @@
+    --------
+    // Standard Usage:
+    > var cx;
+    > cx = new {{alias:@stdlib/array/complex64}}( [ -2.0, 1.0, 3.0, -5.0, 4.0, 0.0 ] );
```
same comment as above.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/types/index.d.ts:
```diff
@@ -0,0 +1,96 @@
+	* var icx = icamax( cx.length, cx, 1 );
+	* // returns 1
+	*/
+	( N: number, zx: Complex64Array, strideX: number ): number;
```
It should be `cx` instead of `zx`.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/types/index.d.ts:
```diff
@@ -0,0 +1,96 @@
+	* var icx = icamax.ndarray( cx.length, cx, 1, 0 );
+	* // returns 1
+	*/
+	ndarray( N: number, zx: Complex64Array, strideX: number, offsetX: number ): number;
```
same comment as above.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/types/index.d.ts:
```diff
@@ -0,0 +1,96 @@
+* Finds the index of the first element having maximum |Re(.)| + |Im(.)|
+*
+* @param N - number of indexed elements
+* @param zx - input array
```
Same comment as above. Change `zx` to `cx` here and elsewhere.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/lib/icamax.js:
```diff
@@ -0,0 +1,53 @@
+/**
+* Finds the index of the first element having maximum |Re(.)| + |Im(.)|.
+*
+* @param {PositiveInteger} N - number of indexed elements
+* @param {Float64Array} x - input array
+* @param {integer} strideX - `x` stride length
+* @returns {integer} index value
```
Stick to the variables you assign here and elsewhere they must be same. Either keep them as `x` everywhere or `cx` everywhere.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/lib/index.js:
```diff
@@ -0,0 +1,68 @@
+*
+* var cx = new Complex64Array( [ -2.0, 1.0, 3.0, -5.0, 4.0, 0.0, -1.0, -3.0 ] );
+*
+* var icx = icamax( cx.length, cx, 1, 0 );
```
It should be `icamax.ndarray()`;

aman-095 reviewed (commented):
Thanks for working on this. Left an initial set of comments.

ShabiShett07 committed:
fix: resolved errors

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

ShabiShett07 committed:
fix: resolved errors

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

ShabiShett07 commented:
Hello @aman-095 sir, thank you for suggesting the changes and made the necessary changes. Please do check.

ShabiShett07 commented:
cc: @kgryte @Planeshifter

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/icamax

ShabiShett07 committed:
chore: update implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: update variable name

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: update implementation

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: update implementation

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: add ndarray example

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/icamax

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented on lib/node_modules/@stdlib/blas/base/icamax/README.md:
````diff
@@ -0,0 +1,172 @@
+// returns 1
+```
+
+#### icamax.ndarray( N, x, strideX, offset )
````
```suggestion
#### icamax.ndarray( N, x, strideX, offsetX )
```

ShabiShett07 commented on lib/node_modules/@stdlib/blas/base/icamax/README.md:
```diff
@@ -0,0 +1,172 @@
+
+-   **offsetX**: starting index.
+
+While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the `offset` parameter supports indexing semantics based on a starting index. For example, to start from the second index,
```
```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameter supports indexing semantics based on a starting index. For example, to start from the second index,
```

ShabiShett07 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/repl.txt:
```diff
@@ -0,0 +1,89 @@
+
+{{alias}}( N, x, stride )
```
```suggestion
{{alias}}( N, x, strideX )
```

ShabiShett07 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/repl.txt:
```diff
@@ -0,0 +1,89 @@
+{{alias}}( N, x, stride )
+    Finds the index of the first element having maximum |Re(.)| + |Im(.)|.
+
+    The `N` and `stride` parameters determine which elements in `x` are
```
```suggestion
    The `N` and `strideX` parameters determine which elements in `x` are
```

ShabiShett07 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/repl.txt:
```diff
@@ -0,0 +1,89 @@
+    x: Complex64Array
+        Input array.
+
+    stride: integer
```
```suggestion
    strideX: integer
```

ShabiShett07 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/repl.txt:
```diff
@@ -0,0 +1,89 @@
+    > var y = {{alias}}( x.length, x, 1 )
+    1
+
+    // Using `N` and `stride` parameters:
```
```suggestion
    // Using `N` and `strideX` parameters:
```

ShabiShett07 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/repl.txt:
```diff
@@ -0,0 +1,89 @@
+    1
+
+
+{{alias}}.ndarray( N, x, stride, offsetX )
```
```suggestion
{{alias}}.ndarray( N, x, strideX, offsetX )
```

ShabiShett07 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/repl.txt:
```diff
@@ -0,0 +1,89 @@
+    x: Complex64Array
+        Input array.
+
+    stride: integer
```
```suggestion
    strideX: integer
```

ShabiShett07 committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 commented:
@kgryte, made some clean-up which I found, need your review

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/README.md:
```diff
@@ -0,0 +1,172 @@
+
+var x = new Complex64Array( [ -2.0, 1.0, 3.0, -5.0, 4.0, 0.0, -1.0, -3.0 ] );
+
+var y = icamax( x.length, x, 1 );
```
can use `idx` instead of `y` here and elsewhere.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/README.md:
```diff
@@ -0,0 +1,172 @@
+// Create an offset view:
+var x1 = new Complex64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element
+
+// Find index of element having the maximum absolute value:
```
instead to `maximum absolute value` update this to -> `maximum |Re(.)| + |Im(.)|.`

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/benchmark/benchmark.js:
```diff
@@ -0,0 +1,104 @@
+	* @param {Benchmark} b - benchmark instance
+	*/
+	function benchmark( b ) {
+		var y;
```
replace `y` with `idx` here and elsewhere across this PR.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/docs/types/index.d.ts:
```diff
@@ -0,0 +1,96 @@
+	*
+	* @param N - number of indexed elements
+	* @param x - input array
+	* @param stride - stride length for `x`
```
Maintain consistency across either use `stride` or use `strideX`.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/examples/index.js:
```diff
@@ -0,0 +1,38 @@
+	return new Complex64( discreteUniform( 0, 10 ), discreteUniform( -5, 5 ) );
+}
+
+// Generate random input arrays:
```
```suggestion
// Generate random input array:
```

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/lib/icamax.js:
```diff
@@ -0,0 +1,53 @@
+*/
+function icamax( N, x, strideX ) {
+	var ox = stride2offset( N, strideX );
+	return ndarray( N, x, strideX, ox );
```
```suggestion
	return ndarray( N, x, strideX, stride2offset( N, strideX ) );
```

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/lib/index.js:
```diff
@@ -0,0 +1,68 @@
+*
+* var y = new Float64Array( [ -2.0, 1.0, 3.0, -5.0, 4.0, 0.0, -1.0, -3.0 ] );
+*
+* var y = icamax( y.length, y, 1 );
```
```suggestion
* var x = new Float64Array( [ -2.0, 1.0, 3.0, -5.0, 4.0, 0.0, -1.0, -3.0 ] );
*
* var idx = icamax( x.length, x, 1 );
```

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/lib/index.js:
```diff
@@ -0,0 +1,68 @@
+*
+* var y = new Complex64Array( [ -2.0, 1.0, 3.0, -5.0, 4.0, 0.0, -1.0, -3.0 ] );
+*
+* var y = icamax.ndarray( y.length, y, 1, 0 );
```
same as above.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/lib/ndarray.js:
```diff
@@ -0,0 +1,75 @@
+* // returns 1
+*/
+function icamax( N, x, strideX, offsetX ) {
+	var max;
+	var ix;
+	var v;
+	var y;
+	var i;
```
replace y with idx.

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/lib/ndarray.js:
```diff
@@ -0,0 +1,75 @@
+	max = scabs1( x.get( offsetX ) );
+	ix = offsetX + strideX;
+	for ( i = 1; i < N; i++ ) {
+		v = scabs1( x.get( ix ) );
+		if ( v > max ) {
+			y = i;
+			max = v;
```
```suggestion
		if ( scabs1( x.get( ix ) ) > max ) {
			y = i;
			max = scabs1( x.get( ix ) );
```

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/package.json:
```diff
@@ -0,0 +1,75 @@
+    "level 1",
+    "icamax",
+    "maximum",
+    "dcabs1",
```
```suggestion
    "scabs1",
```

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/test/test.icamax.js:
```diff
@@ -0,0 +1,182 @@
+	x = new Complex64Array( [ 3.0, -4.0, 1.0, 15.0, 4.0, 3.0 ] );
+	expected = 1;
+
+	y = icamax( x.length, x, -1 );
```
```suggestion
	y = icamax( 3, x, -1 );
```

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/test/test.ndarray.js:
```diff
@@ -0,0 +1,202 @@
+	t.end();
+});
+
+tape( 'the function supports accessing elements in reverse order', function test( t ) {
```
```suggestion
tape( 'the function supports specifying a negative stride', function test( t ) {
```

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/test/test.ndarray.js:
```diff
@@ -0,0 +1,202 @@
+
+	// eslint-disable-next-line max-len
+	x = new Complex64Array( [ 0.1, 4.0, 999.0, 999.0, -0.3, 6.0, 999.0, 999.0, -0.5, 7.0, 999.0, 999.0, -0.1, 3.0 ] );
+	y = icamax( 4, x, -2, x.length-1 );
```
```suggestion
	y = icamax( 4, x, -2, 6 );
```

aman-095 commented on lib/node_modules/@stdlib/blas/base/icamax/test/test.ndarray.js:
```diff
@@ -0,0 +1,202 @@
+	y = icamax( x.length, x, -1, x.length-1 );
+	t.strictEqual( y, 1, 'returns expected value' );
+
+	y = icamax( 2, x, -1, x.length-2 );
```
```suggestion
	y = icamax( 2, x, -1, 1 );
```

aman-095 reviewed (commented):
@ShabiShett07 Left an initial set of comments. Once addressed, will be ready to merge.

aman-095 commented:
@ShabiShett07, before resolving any comment make sure the change is incorporated incase of any issues mention there itself. It helps later reviewing the PR again.

ShabiShett07 committed:
chore: change y to idx

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: add appropriate statements

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
test: add negative stride

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
docs: update jsdoc

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: update comment

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: reducing variable

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: add appropriate keyword

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
test: remove duplicate

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
refactor: reduce variable

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: add appropriate parameter value

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
@aman-095, sorry for that mistake, I have checked the changes I made now, need your review

kgryte committed:
refactor: avoid re-materializing complex number instances

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
fix: address parentheses

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/icamax

kgryte commented on lib/node_modules/@stdlib/blas/base/icamax/lib/index.js:
```diff
@@ -0,0 +1,68 @@
+* var Complex64Array = require( '@stdlib/array/complex64' );
+* var icamax = require( '@stdlib/blas/base/icamax' );
+*
+* var x = new Float64Array( [ -2.0, 1.0, 3.0, -5.0, 4.0, 0.0, -1.0, -3.0 ] );
```
```suggestion
* var x = new Complex64Array( [ -2.0, 1.0, 3.0, -5.0, 4.0, 0.0, -1.0, -3.0 ] );
```

kgryte committed:
docs: fix constructor name

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/icamax

MeKaustubh07 committed:
chore: add string format in bench

kgryte committed:
Merge branch 'develop' into feature/icamax

kgryte commented:
/stdlib update-copyright-years

kgryte commented:
@MeKaustubh07 Would you mind doing a review of this, if you haven't already?

stdlib-bot committed:
chore: update copyright years

MeKaustubh07 commented:
> @MeKaustubh07 Would you mind doing a review of this, if you haven't already?

@kgryte I'm currently working through it locally. The review request notifications are being triggered automatically whenever I push trial commits while working on someone else's branch, so sorry about the noise.

I'll make sure to explicitly mention you when I've finished reviewing and the PR is ready for your attention.

kgryte commented:
@MeKaustubh07 Sounds good. Obviously I just merged in the latest and updated the copyright years, so you'll want to sync your local copy with the current remote.

MeKaustubh07 committed:
docs: fix variable naming and description

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: na
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

MeKaustubh07 committed:
refactor: example acc to scasum

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: na
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

MeKaustubh07 committed:
chore: format and use correct type

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: na
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

MeKaustubh07 committed:
fix: remove unwanted spaces in test

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: na
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

MeKaustubh07 committed:
chore: cleanup

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: passed
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

MeKaustubh07 commented:
@kgryte ready for the review.

MeKaustubh07 commented on lib/node_modules/@stdlib/blas/base/icamax/test/test.icamax.js:
```diff
@@ -0,0 +1,180 @@
+	expected = 2;
+
+	idx = icamax( 3, x1, 1 );
+	t.strictEqual( idx, expected, 'returns expected value' );
```
```suggestion
	t.strictEqual( idx, expected, 'returns expected value' );
	
```

MeKaustubh07 commented on lib/node_modules/@stdlib/blas/base/icamax/test/test.icamax.js:
```diff
@@ -0,0 +1,180 @@
+	expected = 1;
+
+	idx = icamax( 2, x, 2 );
+	t.strictEqual( idx, expected, 'returns expected value' );
```
```suggestion
	t.strictEqual( idx, expected, 'returns expected value' );
	
```

MeKaustubh07 commented on lib/node_modules/@stdlib/blas/base/icamax/README.md:
```diff
@@ -0,0 +1,172 @@
+
+-   **offsetX**: starting index.
+
+While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameter supports indexing semantics based on a starting index. For example, to start from the second element.
```
```suggestion
While [`typed array`][mdn-typed-array] views mandate a view `offset` based on the underlying buffer, the offset parameter supports indexing semantics based on a starting index. For example, to start from the second element.
```

MeKaustubh07 commented on lib/node_modules/@stdlib/blas/base/icamax/test/test.ndarray.js:
```diff
@@ -0,0 +1,200 @@
+	expected = 2;
+
+	idx = icamax( 3, x, 1, 1 );
+	t.strictEqual( idx, expected, 'returns expected value' );
```
```suggestion
	t.strictEqual( idx, expected, 'returns expected value' );
	
```

MeKaustubh07 commented on lib/node_modules/@stdlib/blas/base/icamax/test/test.ndarray.js:
```diff
@@ -0,0 +1,200 @@
+	expected = 1;
+
+	idx = icamax( 2, x, 2, 1 );
+	t.strictEqual( idx, expected, 'returns expected value' );
```
```suggestion
	t.strictEqual( idx, expected, 'returns expected value' );
	
```

MeKaustubh07 committed:
Apply suggestions from code review

Co-authored-by: Aman Bhansali <92033532+aman-095@users.noreply.github.com>
Signed-off-by: Kaustubh Patange <kaustubh.mp007@gmail.com>

MeKaustubh07 committed:
Update lib/node_modules/@stdlib/blas/base/icamax/test/test.icamax.js

Signed-off-by: Kaustubh Patange <kaustubh.mp007@gmail.com>

MeKaustubh07 committed:
Apply suggestions from code review

Co-authored-by: Kaustubh Patange <kaustubh.mp007@gmail.com>
Signed-off-by: Kaustubh Patange <kaustubh.mp007@gmail.com>

MeKaustubh07 committed:
style: fix CRLF and trailing whitespace in icamax

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: na
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

MeKaustubh07 committed:
fix: readme description

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: passed
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---
